apps/idos-enclave/src/pages/methods/Password.tsx

```diff
@@ -1,43 +1,70 @@
+import * as Base64Codec from "@stablelib/base64";
 import { useCallback, useEffect, useRef, useState } from "preact/hooks";
+import nacl from "tweetnacl";
 import { Button } from "../../components/Button";
 import { Heading } from "../../components/Heading";
 import { Input } from "../../components/Input";
 import { Paragraph } from "../../components/Paragraph";
+import { idOSKeyDerivation } from "../../lib/idOSKeyDerivation";
 import type { MethodProps } from "./Chooser";
 
 export default function Password({
   mode,
   onSuccess,
   store,
-}: MethodProps<{ password: string; duration: number }>) {
+  encryptionPublicKey,
+}: MethodProps<{ password: string; duration: number }> & { encryptionPublicKey: string }) {
   const [password, setPassword] = useState("");
   const [duration, setDuration] = useState(7);
   const passwordInput = useRef<{ focus: () => void }>(null);
+  const [error, setError] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  async function derivePublicKeyFromPassword(password: string) {
```

Comment:
What do you think of, instead of duplicating code and imports over here, sending a message back to the iframe enclave with the password string and deriving the key over there?